packages/deployer/subtasks/validate-namespaces.js

```diff
@@ -0,0 +1,49 @@
+const path = require('path');
+const rimraf = require('rimraf');
+const { subtask } = require('hardhat/config');
+const logger = require('@synthetixio/core-js/utils/logger');
+
+const {
+  getContractDataFromDebugFile,
+  getDebugDataFile,
+  getSlotAddress,
+  findDuplicateSlots,
+} = require('../internal/ast-helper');
+const { SUBTASK_VALIDATE_NAMESPACES } = require('../task-names');
+
+subtask(SUBTASK_VALIDATE_NAMESPACES).setAction(async (_, hre) => {
```

Comment:
IMO this subtask should be renamed to "VALIDATE_STORAGE", and it should  very concisely call functions whose names are very declarative as of what is being validated, e.g:
`_ensureNoRegularStorageSlots(...)`
`_ensureNoDuplicateStorageNamespaces(...)`
`_ensureAppendOnlyMutationsOnNamespaces(...)`
etc...

Right now, if you look at this code, it is not clear until you understand the whole thing, exactly what is being validated.